Controls/FAIcon/src/DotVVM.Contrib/Scripts/DotVVM.Contrib.FAIcon.js

```diff
@@ -1,27 +1,28 @@
 ﻿ko.bindingHandlers["dotvvm-contrib-FAIcon"] = {
     init: function (element, valueAccessor, allBindingsAccessor, viewModel, bindingContext) {
-
-        valueAccessor().subscribe(function (previousValue) {
-            const splitterIndex = previousValue.lastIndexOf("_");
-            const previousIconValue = previousValue.substr(0, splitterIndex);
-            const previousIconValueClass = "fa-" + previousIconValue.replace('_', '-')
-            const previousStyle = previousValue.substr(splitterIndex + 1);
-            const previousStyleClass = `fa${previousStyle.charAt(0)}`;
-            element.classList.remove(previousStyleClass)
-            element.classList.remove(previousIconValueClass);
-        }, this, "beforeChange");
     },
     update: function (element, valueAccessor, allBindingsAccessor, viewModel, bindingContext) {
+        removePreviousClasses(element);
         const value = ko.unwrap(valueAccessor());
+        addClassesBasedOnIcon(element, value);
+
+        function removePreviousClasses(element) {
```

Comment:
Here is a theoretical breaking change, because if someone uses on dc:FAIcon class with fa prefix it will be removed on every update.

Reply:
this is fine
if we want to be extra save we can throw exception when user tries to set one of the classes from the outside

Reply:
Your call

Reply:
I will keep it as it is. If the user adds any custom class with `fa*` prefix he will find the problem right away and just replace it with something better.